desktop/apps/auction2/components/filter_sort/index.js

```diff
@@ -0,0 +1,54 @@
+import { default as React, PropTypes } from 'react';
+import { connect } from 'react-redux'
+import _ from 'underscore'
+import { updateSort } from '../../actions'
+
+function FilterSort({ sortMap, filterParams, updateSortAction }) {
+  const selectedSort = filterParams.sort
+  return (
+    <div className='auction2-filter-sort'>
+      <span className='auction2-filter-sort__label bordered-pulldown-label'>Sort by:</span>
+      <div className='bordered-pulldown'>
+        <a className='bordered-pulldown-toggle' href='#'>
+          <span className='bordered-pulldown-text'>{ sortMap[selectedSort] }</span>
+          <div className='bordered-pulldown-toggle-caret'>
+            <span className='caret'></span>
+          </div>
+        </a>
+        <div className='bordered-pulldown-options'>
+          {
+            _.map(sortMap, (sortName, sort) => {
+              const selected = sort == selectedSort
+              return <a
+                href='#'
+                className={ selected ? 'borderd-pulldown-active' : '' }
+                key={sort}
+                onClick={() => updateSortAction(sort)}
+              >{sortName}</a>
+            })
+          }
+        </div>
+      </div>
+    </div>
+  )
+}
+
+const mapStateToProps = (state) => {
+  return {
+    sortMap: state.auctionArtworks.sortMap,
+    filterParams: state.auctionArtworks.filterParams
+  }
+}
+
+const mapDispatchToProps = (dispatch) => {
+  return {
+    updateSortAction: (sort) => dispatch(updateSort(sort))
+  }
+}
```

Comment:
if your action creator looks like

```javascript
function updateSort(sort) {
  ...
}
```

(meaning, you don't need to do any strange intermediate stuff before you dispatch)

Then you can do 
```javascript
const mapDispatchToProps = () => ({
   updateSortAction: updateAction,
   someOtherAction: ...      
})
```

(Note that for objects, if you wrap the return object in parens `=> `({ .. })` you don't need an explicit return statement.)

Another thing about the `mapDispatchToProps` argument is that this works too:

```
connect(mapStateToProps, { 
  actionOne, 
  actionTwo, 
  actionThree 
})(MyComponent)
```

Stuff passed in like that will automatically be wrapped with `dispatch`.